src/wallet/test/coinselector_tests.cpp

```diff
@@ -922,5 +922,52 @@ BOOST_AUTO_TEST_CASE(effective_value_test)
     BOOST_CHECK_EQUAL(output5.GetEffectiveValue(), nValue); // The effective value should be equal to the absolute value if input_bytes is -1
 }
 
+BOOST_AUTO_TEST_CASE(SelectCoins_effective_value_test)
+{
+    // Test that effective value is used in coin selection
+    // This test creates an external coin with a high effective value
+    // The coin will not be selected because its effective value is less than the target
```

Comment:
No coin selection is actually happening, as `m_allow_other_inputs` = false. Being high can sometimes be a problem, but in this case the effective value is too *low*.

```suggestion
    // Test that effective value is used to check the value of preset inputs.
    // This test creates a coin in which the effective value is too low to cover the fees, but is higher than the target value.
    // The coin is selected using coin control, with m_allow_other_inputs = false. SelectCoins should fail.
```

Reply:
The effective value of a UTXO is `effValue = value - fee`, so rather than the effective value being too low to cover the fees, the effective value is too low to create the output after deducting the fees:

```suggestion
    // Test that the effective value is used to check whether preset inputs provide sufficient funds when subtract_fee_outputs is not used.
    // This test creates a coin whose value is higher than the target but whose effective value is lower than the target.
    // The coin is selected using coin control, with m_allow_other_inputs = false. SelectCoins should fail due to insufficient funds.
```